third_party/move/move-model/src/model.rs

```diff
@@ -2310,15 +2322,30 @@ impl GlobalEnv {
                 emitln!(writer, "{}", self.display(&*module_spec));
             }
             for str in module.get_structs() {
-                emitln!(writer, "struct {} {{", str.get_name().display(spool));
-                writer.indent();
-                for fld in str.get_fields() {
-                    emitln!(
-                        writer,
-                        "{}: {},",
-                        fld.get_name().display(spool),
-                        fld.get_type().display(tctx)
-                    );
+                if str.has_variants() {
+                    emitln!(writer, "enum {} {{", str.get_name().display(spool));
+                    writer.indent();
+                    for variant in str.get_variants() {
+                        emit!(writer, "{}", variant.display(spool));
+                        let fields = str.get_fields_of_variant(variant).collect_vec();
```

Comment:
Note that `get_fields_of_variant()` is somewhat expensive, as it will iterate over`field_data` for all variants each time you call it.  You may want to define a more targeted helper function like `get_variants_with_fields()` which can pull data for all variants in one pass.

Reply:
I think this is marginal, we are talking in the majority of cases about like 4 fields or so. Let us not make the code unnecessary complex because of such details.

Reply:
Yeah, I misunderstood the scope of get_fields_of_variant.